src/AbstractAlgebra.jl

```diff
@@ -1241,9 +1241,12 @@ getindex(S::Union{Set, Group}, i::Int) = gen(S, i)
 #
 ################################################################################
 
-typed_hvcat(R::Ring, dims::Dims, d...) = matrix(R, length(dims), dims[1], hvcat(dims, d...))
-typed_hcat(R::Ring, d...) = matrix(R, 1, length(d), hcat(d...))
-typed_vcat(R::Ring, d...) = matrix(R, length(d), 1, vcat(d...))
+VERSION >= v"1.7" && (Base.typed_hvncat(R::NCRing, args...) = _matrix(R, hvncat(args...)))
```

Comment:
`Base.typed_hvncat` is called by the `[;;]` syntax, introduced in the fifth point of [the Julia 1.7 news](https://docs.julialang.org/en/v1.7/NEWS/#New-language-features).